src/Kestrel.Transport.Sockets/Internal/SocketConnection.cs

```diff
@@ -167,6 +168,11 @@ internal SocketConnection(Socket socket, MemoryPool<byte> memoryPool, PipeSchedu
                     _trace.ConnectionReadFin(ConnectionId);
                     break;
                 }
+                else if (bytesReceived == WouldBlock)
+                {
+                    // WouldBlock
```

Comment:
This might be worth logging at the trace level.

Reply:
This should be very rare. The OS says: data is available, and then when we try to receive it says: try again.

Reply:
I'm glad we have someone who has experience with this stuff working on this. It looked to me like we were just rapidly spinning while waiting for more data 🤣